linera-core/src/chain_worker/state/temporary_changes.rs

```diff
@@ -226,19 +214,12 @@ where
         // Verify that all required bytecode hashed certificate values and blobs are available, and no
         // unrelated ones provided.
         self.0
-            .check_no_missing_blobs(
-                block,
-                block.published_blob_ids(),
-                hashed_certificate_values,
-                blobs,
-            )
-            .await?;
-        // Write the values so that the bytecode is available during execution.
-        // TODO(#2199): We should not persist anything in storage before the block is confirmed.
```

Comment:
Did we solve this TODO?

Reply:
We're replacing the hashed certificate values with blobs, and since blobs don't write anything to storage before the block is confirmed, then yes! It does fix it